src/v/security/license.cc

```diff
@@ -185,7 +185,7 @@ void parse_data_section_v1(license& lc, const json::Document& doc) {
     if (lc.organization == "") {
         throw license_invalid_exception("Cannot have empty string for org");
     }
-    lc.type = integer_to_license_type(doc.FindMember("type")->value.GetInt());
+    lc._type_str = doc.FindMember("type")->value.GetString();
```

Comment:
`lc._type` should be set where possible, for backwards compatibility (if it's re-serialized)

Reply:
i.e. add a test to ensure that v0 enterprise -> v1 -> v0 is still enterprise.

Reply:
Oh right, license version 1 is rejected by older brokers, so having compatibility of the serde format is not required.